src/utils/streams.js

```diff
@@ -0,0 +1,118 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+// @flow
+
+// This file holds various utilities about streams.
+
+import { Transform, Writable } from 'stream';
+import crypto from 'crypto';
+
+import { getLogger, type Logger } from '../log';
+
+/**
+ * This transform computes a sha1 of the daa passing through it, but otherwise
+ * doesn't alter the data.
+ */
+export class HasherPassThrough extends Transform {
+  log: Logger = getLogger('HasherPassThrough');
+  sha1Hasher: crypto$Hash = crypto.createHash('sha1');
```

Comment:
Yeah I think I can import the type `Hash` from `crypto`. Flow has a lot of these builtin types, they're usually `<node library>$<type>`, I found them useful to use when I was writing the code because they didn't require me to add the import :-) But I can add the `import` now.

Reply:
plot twist: it's not available as a normal exported type. I'll add a comment.

Reply:
actually I can just remove the typing, and flow seems to figure it out just fine.